### PR TITLE
Simplify logging and get rid of log singleton

### PR DESCRIPTION
**Pull request by StefanHauth**
# Description
This PR is a preparation for running the troubleshoot command within the support archive command.
First, it gets rid of the hidden log singleton in troubleshoot. This is needed so that the support archive command can provide an own logger to grab all output into a file. With this, logging had to be reworked a bit to keep separate subtest loggers with dedicated names.

Furthermore, formatting log lines seems to be implemented overly complicated, this has been simplified along the way.

## How can this be tested?
Build the operator, deploy it and run the troubleshoot command. Everything should work as before.

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: src/cmd/troubleshoot/image_test.go [hunk 1]
```diff
@@ -1,16 +1,17 @@
 package troubleshoot
```
Comment (0sewa0) [inline src/cmd/troubleshoot/image_test.go:1]: this amount of tests feel a bit overkill 😅

**Review by 0sewa0 — COMMENTED** [review]
To be honest I reeeally dislike the "style" of this package, I could complain sooo much.

BUT,  the problem is the `troubleshootContext`, which is not new.

The new addition for logging is fine, and its a minimal approach to make it work, without reworking the whole thing.

TLDR: It is fine, but we should rework the whole package later. (pls fix linter 😄 )

**Comment by StefanHauth** [comment]
> To be honest I reeeally dislike the "style" of this package, I could complain sooo much.
> 
> BUT, the problem is the `troubleshootContext`, which is not new.
> 
> The new addition for logging is fine, and its a minimal approach to make it work, without reworking the whole thing.
> 
> TLDR: It is fine, but we should rework the whole package later. (pls fix linter/test 😄 )

Full agree, I actually started this on another branch already before switching to the low-hanging-fruit approach. I'm planning on initiating the rework.

File: src/cmd/troubleshoot/testutils.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package troubleshoot
```
Comment (0sewa0) [inline src/cmd/troubleshoot/testutils.go:1]: does GO know that this is only for testing? (so that its not built into the actual image)

I know that the `something_test.go` syntax does that, dunno about this

File: src/cmd/troubleshoot/image_test.go (hunk 1 above)
Comment (StefanHauth) [reply src/cmd/troubleshoot/image_test.go:1]: I didn't add any, just compressed them a wee bit.

File: src/cmd/troubleshoot/testutils.go (hunk 2 above)
Comment (0sewa0) [reply src/cmd/troubleshoot/testutils.go:1]: checked, if these functions are only used in tests, then rename the file to `utils_test.go`

Comment (StefanHauth) [reply src/cmd/troubleshoot/testutils.go:1]: As long as those methods are not used in production code, they won't be part of it. The linker should take care of this, just checked it (`strings -6 operator-binary  | grep getNullLogger`).
`utils_test.go` IMHO is misleading as it suggests to contain "test for utils" not "utils for test".

Comment (0sewa0) [reply src/cmd/troubleshoot/testutils.go:1]: could be in `logger_test.go` as it has to do with the logger (and then we dont use the forbidden `utils` word 😛 )

**Review by 0sewa0 — APPROVED** [review]
